source/parsed_grid_generator.cc

```diff
@@ -232,61 +232,138 @@ ParsedGridGenerator<dim, spacedim>::serial()
+              std::vector<unsigned int> un_int_vec_option_one)
+  {
+    if (grid_name == "unit_hyperball")
+      {
+
+        GridGenerator::hyper_ball( tria,
+                                   point_option_one,
+                                   double_option_one);
+      }
+    else if (grid_name == "subhyperrectangle")
+      {
+
+        std::vector<unsigned int> refinement(dim);
+        for (unsigned int i=0; i<dim; ++i)
+          refinement[i] = un_int_option_one;
+        GridGenerator::subdivided_hyper_rectangle ( tria,
+                                                    refinement,
+                                                    point_option_two,
+                                                    point_option_one);
+      }
+    else if (grid_name == "hyper_shell")
+      {
+        GridGenerator::hyper_shell( tria,
+                                    point_option_one,
+                                    double_option_two,
+                                    double_option_one,
+                                    un_int_option_one,
+                                    bool_option_one);
+      }
+  }
+}
+
 template <int dim, int spacedim>
 void ParsedGridGenerator<dim, spacedim>::create(Triangulation<dim,spacedim> &tria)
 {
   Assert(grid_name != "", ExcNotInitialized());
-  if (grid_name == "rectangle")
-    {
-      GridGenerator::subdivided_hyper_rectangle (tria, un_int_vec_option_one,
-                                                 point_option_two, point_option_one,
-                                                 bool_option_one);
-    }
-  // TO BE DONE WHEN POSSIBLE
-  // else if(grid_name == "unit_hyperball")
-  // {
-  //   Point<spacedim> center;
-  //   double radius=1.;
-  //
-  //   GridGenerator::hyper_ball (tria,
-  //                               center, radius);
-  //   }
-  // else if(grid_name == "subhyperrectangle"){
-  //   if(dim == spacedim)
-  //   {
-  //
-  //   std::vector<unsigned int> refinement(spacedim);
-  //   for(unsigned int i=0; i<spacedim;++i)
-  //     refinement[i] = un_int_option_one;
-  //   GridGenerator::subdivided_hyper_rectangle (tria, refinement,
-  //                                    point_option_two, point_option_one);
-  //   }
-  //   else
-  //     Assert(true, ExcInternalError("dim != spacedim not supported"))
-  //
-  // }
-  else if (grid_name == "file")
-    {
-      GridIn<dim, spacedim> gi;
-      gi.attach_triangulation(tria);
-
-      std::ifstream in(input_grid_file_name.c_str());
-      AssertThrow(in, ExcIO());
 
-      std::string ext = extension(input_grid_file_name);
-      if (ext == "vtk")
-        gi.read_vtk(in);
-      else if (ext == "msh")
-        gi.read_msh(in);
-      else if (ext == "ucd" || ext == "inp")
-        gi.read_ucd(in);
-      else if (ext == "unv")
-        gi.read_unv(in);
-      else
-        Assert(false, ExcNotImplemented());
-    }
-  else
-    Assert(false, ExcInternalError("Unrecognized grid."));
```

Comment:
@luca-heltai I deleted this assert. If the grid name is not correct the parser should return an error and if dim is incompatible with spacedim an assert of deal.ii should stop the program.